### ISSUE TITLE
DedicatedLoop Is now ignored

### ISSUE THREAD

**Issue by Papi94** [OpenRA/OpenRA#14069]
Server will continuously loop regardless of what the DedicatedLoop boolean is set as

**Comment by pchote** [comment]
The DedicatedLoop argument was removed in release-20160508 when the dedicated server was split from the main game executable.

**Comment by daps94** [comment]
I would like to use this option so that i can call a log parser after each game completes so that i can record games and statistics on a public server. I've spoken to other server admins and they are on board. Could we please get this feature back pchote?

**Comment by pchote** [comment]
Your best bet would be to implement your stats logic in your own `ServerTrait` class.  If you implement `IEndGame` then the server will automatically notify you when the game has ended.

**Comment by daps94** [comment]
Thank you Pchote. Unfortunately im mostly a php programmer and c# is a bit over my head.

Could you dumb it down for me? Im a quick learn and i've been reading up on c# for the hour or so.

To be specific,

Where would i write my servertrait class,, in Server.cs , or Traitinterfaces.cs?

Do i have access to player scores? If not, whats the best way to determine the winner of the game?

If possible i'd like to send the following data to the central api, for stats tracking and user authentication

players handles
players ips
players scores
player teams


The api will be a simple get api, a request would look something like

Ex

http://openrastats.com/api.php?hostauthentication=dsg4bi3uhh5iweht9243h456266245&handles=Papi||Apetit||Abbart||wee man&ips=1.2.3.4-5.6.7.8-8.7.6.5-1.32.5.7&scores=1231-52343-2342-22&teams=1-1-2-2

Then each array of data would be processed server side

Every server hoster participating, will get there own api key so that i can verify that those sending me data are legit, ill need some way to define this, ideally when i call the server as an argument.

**Comment by abcdefg30** [comment]
Looping can be done via a script. We have other issues covering the stats aspect.
